### ISSUE TITLE
Adding a vuex store using a module

### ISSUE THREAD

**Issue by sebastianmacias** [nuxt/nuxt.js#1798]
Is there a way to add a vuex store in "Modules mode" using [Nuxt modules](https://nuxtjs.org/guide/modules)? 

I couldn't find any info in the documentation or in the source of the community modules.

**Comment by alexchopin** [comment]
Hi, 
You can add a vuex store with a plugin and use `addPlugin()`.
We are still working on the documentation. (WIP)

**Comment by chihaoyo** [comment]
Hi @sebastianmacias @alexchopin. I’m building an several separate nuxt.js projects and I want to package my vuex module for it to be used in all of these nuxt.js projects, so they all start with the same vuex state, mutations, actions, etc. 

I’ve been looking for a way to do this and found my way to this issue. Are there any progress in adding the documentation you mentioned? **OR** is there any way I can help with this? **OR** is there another way to achieve what I want to do? Thanks!

**Comment by silverbackdan** [comment]
@chihaoyo I stumbled across this thread and am looking to do similar a thing at the moment myself. I'm still at the early stages of my research but I've found looking at the nuxt-community modules are a very helpful guide. Specifically the axios one https://github.com/nuxt-community/axios-module/tree/master/lib

You'll see in there the module adds a plugin. In plugins, the context makes the store available including getters, dispatch etc.

This guided me to the solution of creating the store from a plugin:
https://github.com/nuxt-community/modules/issues/97

Which references the auth module:
https://github.com/nuxt-community/auth-module

This line registers a new vuex store module:
https://github.com/nuxt-community/auth-module/blob/dev/lib/core/storage.js#L87

I'm hoping to put all of this into practice in my own modules too, but my advice with reusable modules is to follow @pi0 and the nuxt-community modules - his code and these modules have taught me more than any docs could IMO.

Sorry if this is a too late - but I hope it helps.
